iam/src/providers/fiveOrMoreGithubRepos.ts

```diff
@@ -0,0 +1,92 @@
+// ----- Types
+import type { RequestPayload, VerifiedPayload } from "@gitcoin/passport-types";
+import type { Provider, ProviderOptions } from "../types";
+
+// ----- HTTP Client
+import axios from "axios";
+
+export type GithubTokenResponse = {
+  access_token: string;
+};
+
+export type GithubFindMyUserResponse = {
+  id?: string;
+  login?: string;
+  public_repos?: number;
+  type?: string;
+};
+
+// Export a Github Provider to carry out OAuth, check if the user has 5 >= repos,
+// and return a record object
+export class FiveOrMoreGithubRepos implements Provider {
+  // Give the provider a type so that we can select it with a payload
+  type = "FiveOrMoreGithubRepos";
+
+  // Options can be set here and/or via the constructor
+  _options = {};
+
+  // construct the provider instance with supplied options
+  constructor(options: ProviderOptions = {}) {
+    this._options = { ...this._options, ...options };
+  }
+
+  // verify that the proof object contains valid === "true"
+  async verify(payload: RequestPayload): Promise<VerifiedPayload> {
+    let valid = false,
+      verifiedPayload: GithubFindMyUserResponse = {};
+
+    try {
+      verifiedPayload = await verifyGithubRepoCount(payload.proofs.code);
+    } catch (e) {
+      return { valid: false };
+    } finally {
+      valid = verifiedPayload && verifiedPayload.public_repos >= 5 && verifiedPayload.id ? true : false;
+    }
+
+    return {
+      valid: valid,
+      record: {
+        id: verifiedPayload.id,
+        repoCount: `${verifiedPayload.public_repos}`,
```

Comment:
It might be better to have this as a statement that matches the range rather than the number of repos that the user has - if the user creates a new repo, they would get a new hash, and would be able to use that new stamp against a different Passport.